lib/storage/src/content_manager/errors.rs

```diff
@@ -42,16 +42,54 @@ impl From<CollectionError> for StorageError {
             CollectionError::Cancelled { description } => StorageError::ServiceError {
                 description: format!("Operation cancelled: {description}"),
             },
-            err @ CollectionError::InconsistentFailure { .. } => StorageError::ServiceError {
-                description: format!("{err}"),
-            },
+            CollectionError::InconsistentShardFailure { ref first_err, .. } => {
+                let full_description = format!("{}", &err);
+                StorageError::from((*first_err.clone(), full_description))
+            }
             CollectionError::BadShardSelection { description } => {
                 StorageError::BadRequest { description }
             }
         }
     }
 }
 
+/// Used to override the `description` field of the resulting `StorageError`
+impl From<(CollectionError, String)> for StorageError {
+    fn from(value: (CollectionError, String)) -> Self {
+        let (err, overriding_description) = value;
+        match err {
+            CollectionError::BadInput { .. } => StorageError::BadInput {
+                description: overriding_description,
+            },
+            CollectionError::NotFound { .. } => StorageError::NotFound {
+                description: overriding_description,
+            },
+            CollectionError::PointNotFound { .. } => StorageError::NotFound {
+                description: overriding_description,
+            },
+            CollectionError::ServiceError { .. } => StorageError::ServiceError {
+                description: overriding_description,
+            },
+            CollectionError::BadRequest { .. } => StorageError::BadRequest {
+                description: overriding_description,
+            },
+            CollectionError::Cancelled { .. } => StorageError::ServiceError {
+                description: format!("Operation cancelled: {overriding_description}"),
+            },
+            CollectionError::InconsistentShardFailure { ref first_err, .. } => {
+                log::warn!(
+                    "No recursive InconsistentShardFailure failure expected {}",
+                    &err
+                );
```

Comment:
It can happen sometimes, when information about shards is not fully propagated - for example after transfer - and two or more remote jumps are done

Reply:
I will remove the warning then!